In [2]:
import requests
import pandas as pd

In [28]:
api_key = "5ZHM042IFQM5PWUB"
symbol = 'IBM'
url = f'https://www.alphavantage.co/query?function=TIME_SERIES_DAILY&symbol={symbol}&apikey={api_key}&datatype=json'

response = requests.get(url)
data = response.json()

In [30]:
stock_data = []
for date, daily_data in data['Time Series (Daily)'].items():
    stock_data.append({
        'date': date,
        'open': daily_data['1. open'],
        'high': daily_data['2. high'],
        'low': daily_data['3. low'],
        'close': daily_data['4. close'],
        'volume': daily_data['5. volume']
    })
df = pd.DataFrame(stock_data)
df['close'] = pd.to_numeric(df['close'], errors='coerce')
df

date      open      high       low   close   volume
0   2024-03-21  193.0000  193.3700  190.0100  191.90  6013561
1   2024-03-20  192.8700  193.9800  191.3100  193.96  3238643
2   2024-03-19  191.4900  193.5800  190.2800  193.34  5317341
3   2024-03-18  191.7000  193.2300  190.3200  191.69  5410562
4   2024-03-15  191.9900  193.0573  190.7000  191.07  8828184
..         ...       ...       ...       ...     ...      ...
95  2023-11-02  145.7700  147.1000  144.8400  147.01  3902657
96  2023-11-01  145.0000  146.5100  144.4500  145.40  4750081
97  2023-10-31  143.0000  144.7600  142.5900  144.64  6592041
98  2023-10-30  143.1900  144.5000  142.5800  142.63  4204190
99  2023-10-27  143.6200  144.7000  141.7100  142.52  5469227

[100 rows x 6 columns]

In [31]:
average_close = df['close'].head(10).mean()
average_close

193.755

Средняя цена закрытия акций за последние 10 дней: 193 доллара

In [32]:
average_close = df['close'].tail(10).mean()
average_close

146.25500000000002

3 месяца назад средняя цена закрытия акций было 146
За эти 3 месяца она выросла приблизительно на 50 долларов